torch/_functorch/aot_autograd.py

```diff
@@ -1858,22 +1858,39 @@ def backward(ctx, *flat_args):
                 list(ctx.symints) + list(ctx.saved_tensors) + list(contiguous_args)
             )
             del contiguous_args
-            if CompiledFunction.compiled_bw is None:
-                # TODO - pass in fake tensors ?
-                context = disable_autocast_manager if disable_amp else nullcontext
-                with context(), track_graph_compiling(aot_config, "backward"):
-                    CompiledFunction.compiled_bw = aot_config.bw_compiler(
-                        bw_module, all_args
+
+            class CompiledFunctionBackward(torch.autograd.Function):
```

Comment:
Just a thought - since this code is hot-path, would it make sense only generate the custom function when we see that the inputs to the backward pass require gradients?

Reply:
Good idea. I also realize that maybe_clear_saved_tensors is referring to the wrong ctx now, so that should fix that also.

Reply:
Updated here: https://github.com/pytorch/pytorch/pull/92688